client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js

```diff
@@ -3,30 +3,29 @@ import { dispatch, getState } from '../../../store'
 import { clearWaiting, setWaiting } from '../../../slices/waiting'
 import { showError } from '../../../slices/error'
 
-export function toggleVolunteerStatus(
+export async function toggleVolunteerStatus(
   eventId,
   willVolunteer,
   setVolunteering,
   consume = requestor
 ) {
-  const storeState = getState()
-  const { id, token } = storeState.user
-  if (!id) {
-    dispatch(showError('Please register or sign in to volunteer.'))
-  } else {
-    dispatch(setWaiting())
-    const routeMethod = willVolunteer ? 'post' : 'delete'
-    const userData = { userId: id, eventId }
+  try {
```

Comment:
Nice work, only a few changes needed! We only wrap a try...catch block around our async calls, any code before the await needs to be outside of the try block.

Reply:
Thanks! Have fixed it now, let us know if its all good :)